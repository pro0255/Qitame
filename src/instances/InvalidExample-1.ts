export const INVALID_EXAMPLE_MINUS_ONE = [100, 45, 21, 60, -1];
