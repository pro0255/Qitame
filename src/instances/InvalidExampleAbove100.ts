export const INVALID_EXAMPLE_ABOVE_100 = [120, 45, 21, 60, -1];
