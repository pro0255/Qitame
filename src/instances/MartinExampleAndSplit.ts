export const MARTIN_EXAMPLE_AND_SPLIT = [46, 45, 21, 60, 24];
