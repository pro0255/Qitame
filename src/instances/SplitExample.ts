export const SPLIT_EXAMPLE = [46];
