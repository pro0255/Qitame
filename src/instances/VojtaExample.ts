export const MARTIN_EXAMPLE = [100, 45, 21, 60, 24];
