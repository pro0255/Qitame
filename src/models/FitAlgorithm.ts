import { Position } from '../types/Position';

export class FitAlgorithm {
  public static fit = (squareEdge: number, contents: number[]): Position[] => {
    const positions: Position[] = [];

    contents.forEach((_) => {
      positions.push({
        x: 1,
        y: squareEdge,
        height: 100,
        width: 100,
      });
    });

    return positions;
  };
}
