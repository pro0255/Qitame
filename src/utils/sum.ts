export const sum = (numbers: number[]): number => {
  return numbers.reduce((accumulator, value) => {
    return accumulator + value;
  }, 0);
};
