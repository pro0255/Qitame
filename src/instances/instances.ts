import { Examples } from '../components/Examples/types';
import { MARTIN_EXAMPLE } from './MartinExample';
import { MARTIN_EXAMPLE_AND_SPLIT } from './MartinExampleAndSplit';
import { SPLIT_EXAMPLE } from './SplitExample';

export const instances: Examples = [
  {
    value: MARTIN_EXAMPLE,
    isValid: true,
    name: "Martin's example",
  },
  {
    value: SPLIT_EXAMPLE,
    isValid: true,
    name: 'Split examples',
  },
  {
    value: MARTIN_EXAMPLE_AND_SPLIT,
    isValid: true,
    name: "Martin's $ split example",
  },
];
