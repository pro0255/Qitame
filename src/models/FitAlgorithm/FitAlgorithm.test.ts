import { FitAlgorithm } from './FitAlgorithm';
import { MARTIN_EXAMPLE } from '../../instances/MartinExample';
import { sum } from '../../utils/sum';

describe('FitAlgorithm', () => {
  it('for the same input, the output remains consistent', () => {
    const rectangleEdges = FitAlgorithm.fit(Math.sqrt(sum(MARTIN_EXAMPLE)), MARTIN_EXAMPLE);

    expect(rectangleEdges).toMatchSnapshot();
  });
});
