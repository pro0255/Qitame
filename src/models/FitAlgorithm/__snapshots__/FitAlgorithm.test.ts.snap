// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FitAlgorithm for the same input, the output remains consistent 1`] = `
[
  {
    "a": 15.811388300841896,
    "b": 6.324555320336759,
  },
  {
    "a": 9.486832980505138,
    "b": 4.743416490252569,
  },
  {
    "a": 11.067971810589327,
    "b": 1.8973665961010278,
  },
  {
    "a": 7.58946638440411,
    "b": 7.905694150420949,
  },
  {
    "a": 3.162277660168378,
    "b": 7.589466384404113,
  },
]
`;
